django/db/models/sql/query.py

```diff
@@ -1065,6 +1059,8 @@ def get_external_cols(self):
     def as_sql(self, compiler, connection):
         sql, params = self.get_compiler(connection=connection).as_sql()
         if self.subquery:
+            if not connections[DEFAULT_DB_ALIAS].features.supports_ordered_subquery:
```

Comment:
You'll want to use provided `connection` here to make sure we support mixed `DATABASES` backends to a certain extent.

Reply:
Also, it must be cleared before calling `get_compiler(connection=connection).as_sql()`.

Reply:
Should've been obvious.. Thanks!